### PR TITLE
Update packaging without azure bdist wheel

### PR DESCRIPTION
**Pull request by lmazuel**
Improve packaging experience:

- Remove hack file azure_bdist_wheel.py
- Delegate nspkg management to azure-storage-common, and remove all mentions in file/queue/blob
- Make azure-storage-nspkg a Python 2 dependency only (Py 3 using PEP420 instead automatically)
- Update tests namespace packages to pkgutil, replace pkg_resources

Adding 3.7 in setup.py as side effect.

@zezha-msft please do not hesitate to double-check everything. I will myself re-double-check and re-post on the PR.

Your CI might need adjustment, because CI is more sensitive than a real installation.

### CODE REVIEW

File: azure-storage-blob/ChangeLog.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 > See [BreakingChanges](BreakingChanges.md) for a detailed list of API breaks.
 
+## Version 1.x.x:
+
+- azure-mgmt-nspkg is not installed anymore on Python 3 (PEP420-based namespace package)
```
Comment (zezha-msft) [inline azure-storage-blob/ChangeLog.md]: Typo, should be `azure-storage-nspkg`.

File: .travis.yml [hunk 2]
```diff
@@ -33,11 +33,11 @@ install:
   - pip install codecov
   - pip install coverage
   - pip install -r requirements.txt
+  - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7"]]; then pip install -e azure-storage-nspkg; fi;  # Would work on Python 3, but will not be installed by PyPI. Trying to be close to real life.
```
Comment (zezha-msft) [inline .travis.yml]: Apparently there's a syntax error:

/home/travis/.travis/job_stages: eval: line 98: syntax error in conditional expression: unexpected token `;'
/home/travis/.travis/job_stages: eval: line 98: syntax error near `;'
/home/travis/.travis/job_stages: eval: line 98: `if [[ "$TRAVIS_PYTHON_VERSION" == "2.7"]]; then pip install -e azure-storage-nspkg; fi; '

File: azure-storage-common/ChangeLog.md [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 - When unable to sign request, avoid wasting time on retries by failing faster.
 - Allow the use of custom domain when creating service object targeting emulators.
+- azure-mgmt-nspkg is not installed anymore on Python 3 (PEP420-based namespace package)
```
Comment (zezha-msft) [inline azure-storage-common/ChangeLog.md]: Typo, should be `azure-storage-nspkg`.

File: azure-storage-file/ChangeLog.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 > See [BreakingChanges](BreakingChanges.md) for a detailed list of API breaks.
 
+## Version 1.x.x:
+
+- azure-mgmt-nspkg is not installed anymore on Python 3 (PEP420-based namespace package)
```
Comment (zezha-msft) [inline azure-storage-file/ChangeLog.md]: Typo, should be azure-storage-nspkg.

File: azure-storage-nspkg/setup.py [hunk 5]
```diff
@@ -43,11 +43,11 @@
         'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
         'License :: OSI Approved :: MIT License',
     ],
     zip_safe=False,
     packages=[
-        'azure',
```
Comment (zezha-msft) [inline azure-storage-nspkg/setup.py]: was this not needed?

File: azure-storage-queue/ChangeLog.md [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 > See [BreakingChanges](BreakingChanges.md) for a detailed list of API breaks.
 
+## Version 1.x.x:
+
+- azure-mgmt-nspkg is not installed anymore on Python 3 (PEP420-based namespace package)
```
Comment (zezha-msft) [inline azure-storage-queue/ChangeLog.md]: Typo, should be azure-storage-nspkg.

File: azure-storage-nspkg/setup.py (hunk 5 above)
Comment (lmazuel) [reply azure-storage-nspkg/setup.py]: because provided by azure-nspkg (this should have been like that in the first place)

**Comment by lmazuel** [comment]
If you build you should see this:
- For azure-storage-nspkg:

   - The sdist should contains 2 files (azure/__init__.py and azure/storage/__init__.py)
   - But the wheel should contain only one (azure/storage/__init__.py)

- For file/queue/blob:

   - The sdist should contains 2 files (azure/__init__.py and azure/storage/__init__.py) and the code files
   - But the wheel should contain only one things under azure/storage/

Also:
- Installing from the wheel in Python 3 should not install any nspkg at all.

In terms of benefits:
- All scenarios of Python 3 works. Whatever mix up PyPI, Github, etc.
- On Python 2, all from Github works, and all from PyPI works. Mix-up from github and from PyPI in the same venv might work depending of the version of setuptools. This is more scenario than right now anyway (thanks to pkgutil).

You can also check this table:
https://github.com/pypa/sample-namespace-packages/blob/master/table.md

You can see than PEP420 + pkgutil is way more compatible than any combinations involving pkg_resource.

**Comment by zezha-msft** [comment]
@lmazuel thank you!! 👍

**Pull request merged**